### PR TITLE
docs: Add deep linking example to user guide

### PR DESCRIPTION
**Pull request by corycaywood**
# Description
Added example in user guide for deep linking to a mini app.

# Checklist
- [x] I have read the [contributing guidelines](CONTRIBUTING.md).
- [x] I have completed the [SDK Development Learning Path](CONTRIBUTING.md#SDK-Development-Learning-Path) (if submitting a feature PR)
- [x] I wrote/updated tests for new/changed code
- [x] I removed all sensitive data before every commit, including API endpoints and keys
- [x] I ran `fastlane ci` without errors

### CODE REVIEW

**Review by Climbatize — APPROVED** [review]
👌

**Review by ts-leojoseph-r — APPROVED** [review]
Looks good to me :+1:

File: USERGUIDE.md [hunk 1]
```diff
@@ -754,6 +754,64 @@ Mini App SDK requires the host app to include the following set of device permis
...
+    }
+
+    let pathComponents = components.path.split("/")
+    guard
+        let rootPath = pathComponents.first
+    else { return false }
+
+    if (rootPath == "miniapp") {
+        guard
+            let id = pathComponents[safeIndex: 1]
```
Comment (laubengaier) [inline USERGUIDE.md:797]: @corycaywood `safeIndex` subscript is a link collection extension which is not provided in Swift per default probably better for the doc to replace the guard statement with just `let id = pathComponents[1]` or something similar

**Review by laubengaier — APPROVED** [review]
LGTM! 👍
minor fix recommendation

File: USERGUIDE.md (hunk 1 above)
Comment (corycaywood) [reply USERGUIDE.md:797]: Thanks, I'll update this.

**Pull request merged**